finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4Message.scala

```diff
@@ -104,9 +104,12 @@ private[h2] object Netty4Message {
       val sz = f.content.readableBytes + f.padding
       val buf = ByteBufAsBuf(f.content.retain())
       val releaser: () => Future[Unit] = {
-        f.content.release()
-        if (sz > 0) () => updateWindow(sz)
-        else () => Future.Unit
+        () =>
+          {
+            val res = if (sz > 0) updateWindow(sz) else Future.Unit
```

Comment:
Take it or leave it, but I think something like
```scala
      val releaser: () => Future[Unit] =
        () =>
          Future { f.content.release(); () }.before {
            if (sz > 0) updateWindow(sz) else Future.Unit
          }
```
might be slightly more idiomatic Scala.